bolt-sidecar/src/primitives/transaction.rs

```diff
@@ -123,15 +130,31 @@ impl std::ops::DerefMut for FullTransaction {
 impl FullTransaction {
     /// Convenience method to parse a raw transaction into a `FullTransaction`.
     pub fn decode_enveloped(data: impl AsRef<[u8]>) -> eyre::Result<Self> {
-        let tx = PooledTransactionsElement::decode_2718(&mut data.as_ref())?;
+        let tx = PooledTransaction::decode_2718(&mut data.as_ref())?;
         Ok(Self { tx, sender: None })
     }
 
     /// Returns the inner transaction.
-    pub fn into_inner(self) -> PooledTransactionsElement {
+    pub fn into_inner(self) -> PooledTransaction {
         self.tx
     }
 
+    /// Returns the signed transaction.
+    pub fn into_signed(self) -> TransactionSigned {
+        match self.tx {
+            PooledTransaction::Legacy(tx) => tx.into(),
+            PooledTransaction::Eip2930(tx) => tx.into(),
+            PooledTransaction::Eip1559(tx) => tx.into(),
+            PooledTransaction::Eip4844(tx) => {
+                let sig = *tx.signature();
+                let hash = *tx.hash();
+                let inner_tx = tx.into_parts().0.into_parts().0;
+                Signed::new_unchecked(inner_tx, sig, hash).into()
+            }
+            _ => unimplemented!(),
```

Comment:
can you remove this? so that if we get a new transaction type we know what we need to update

Reply:
good catch, fixed this two other places as well